### PR TITLE
Add GaussianMixtureInitiator, ASDGaussianInitiator and EnsembleInitiator

### PR DESCRIPTION
**Pull request by csherman-dstl**
Add initiators for GaussianMixture, ASDGaussian and Ensemble state with accompanying tests

### CODE REVIEW

**Comment by csherman-dstl** [comment]
needed to rebase from main

File: stonesoup/initiator/simple.py [hunk 1]
```diff
@@ -294,3 +299,214 @@ def initiate(self, detections, timestamp, **kwargs):
...
+
+    Utilising Gaussian Initiator, sample from the resultant track's state
+    to generate an ASD Gaussian State, overwriting with a
+    :class:`~.ASDGaussianState`.
+    """
+
+    initiator: GaussianInitiator = Property(
+        doc="Gaussian Initiator which will be used to generate tracks.")
+    timestamp: datetime.datetime = Property(
+        doc="Starting timestep attributed to the prior state.")
```
Comment (sdhiscocks) [inline stonesoup/initiator/simple.py:377]: This really shouldn't be required. Can the timestamps be set to list of `None` or I guess list just containing `self.initiator.prior_state.timestamp` like other parameters.

File: stonesoup/initiator/simple.py [hunk 2]
```diff
@@ -294,3 +299,214 @@ def initiate(self, detections, timestamp, **kwargs):
...
+
+        for track in tracks:
+            state = track.state_vector
+            covar = track.covar
+            timestamp = track.timestamp
+
+            track[-1] = ASDGaussianStateUpdate(
+                multi_state_vector=state,
+                timestamps=timestamp,
+                max_nstep=0,
```
Comment (sdhiscocks) [inline stonesoup/initiator/simple.py:421]: I think this should be a parameter defined on the initiator (like `ensemble_size` below).

File: stonesoup/initiator/simple.py [hunk 3]
```diff
@@ -294,3 +299,214 @@ def initiate(self, detections, timestamp, **kwargs):
...
+        for track in tracks:
+            state = track.state_vector
+            covar = track.covar
+            ensemble = self.create_ensemble(state, covar)
+
+            track[-1] = EnsembleStateUpdate(
+                state_vector=ensemble,
+                hypothesis=track.hypothesis,
+                timestamp=track.timestamp)
+
```
Comment (sdhiscocks) [inline stonesoup/initiator/simple.py:511]: Would it be possible to use `EnsembleStateUpdate.from_gaussian_state()` and remove the `create_ensemble` method in the initiator? May need to modify the `from_gaussian_state()` to accept kwargs in order to pass the hypothesis.

**Review by sdhiscocks — CHANGES_REQUESTED** [review]

**Review by sdhiscocks — APPROVED** [review]

**Review by jswright-dstl — APPROVED** [review]

**Pull request merged**